apps/builder/app/builder/features/workspace/canvas-tools/resize-handles.tsx

```diff
@@ -78,59 +77,63 @@ const handleIcon = (
   </svg>
 );
 
-const baseDragProps = {
-  startDistanceThreashold: 0,
-  elementToData: () => true,
-  onStart() {
-    isCanvasPointerEventsEnabledStore.set(false);
-    isResizingCanvasStore.set(true);
-  },
-  onEnd() {
-    isCanvasPointerEventsEnabledStore.set(true);
-    isResizingCanvasStore.set(false);
-  },
+const updateBreakpoint = (width: number) => {
+  const applicableBreakpoint = findApplicableMedia(
+    Array.from(breakpointsStore.get().values()),
+    width
+  );
+  if (applicableBreakpoint) {
+    selectedBreakpointIdStore.set(applicableBreakpoint.id);
+  }
 };
 
-const useResize = () => {
-  const isDragging = useStore(isCanvasPointerEventsEnabledStore) === false;
-  const [isHovering, setIsHovering] = useState(false);
+const useScrub = ({ side }: { side: "right" | "left" }) => {
+  const ref = useRef<HTMLDivElement>(null);
 
-  const updateBreakpoint = (width: number) => {
-    const applicableBreakpoint = findApplicableMedia(
-      Array.from(breakpointsStore.get().values()),
-      width
-    );
-    if (applicableBreakpoint) {
-      selectedBreakpointIdStore.set(applicableBreakpoint.id);
+  useEffect(() => {
+    if (ref.current === null) {
+      return;
     }
-  };
+    const scrub = numericScrubControl(ref.current, {
+      getInitialValue() {
+        return canvasWidthStore.get();
+      },
+      getValue(state, movement) {
+        const value =
+          side === "left"
+            ? state.value - movement * 2
+            : state.value + movement * 2;
+        return Math.max(value, minCanvasWidth);
+      },
+      onStatusChange(status) {
+        if (status === "scrubbing") {
+          isCanvasPointerEventsEnabledStore.set(false);
```

Comment:
mb disable events if isResizingCanvasStore is true instead of adding additional atom?

Reply:
isCanvasPointerEventsEnabledStore is not only for resizing, but also for drag&drop and potentially more interactions, I could automatically set isCanvasPointerEventsEnabledStore when isResizingCanvasStore is true, but I am not sure this would be better, maybe less obvious even